buildpacktoml.go

```diff
@@ -0,0 +1,294 @@
+package lifecycle
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/BurntSushi/toml"
+
+	"github.com/buildpacks/lifecycle/api"
+	"github.com/buildpacks/lifecycle/launch"
+	"github.com/buildpacks/lifecycle/layers"
+)
+
+type BuildTOML struct {
+	BOM   []BOMEntry `toml:"bom"`
+	Unmet []Unmet    `toml:"unmet"`
+}
+
+type Unmet struct {
+	Name string `toml:"name"`
+}
+
+type LaunchTOML struct {
+	BOM       []BOMEntry
+	Labels    []Label
+	Processes []launch.Process `toml:"processes"`
+	Slices    []layers.Slice   `toml:"slices"`
+}
+
+type DefaultBuildpackTOML struct {
+	API       string         `toml:"api"`
+	Buildpack BuildpackInfo  `toml:"buildpack"`
+	Order     BuildpackOrder `toml:"order"`
+	Path      string         `toml:"-"`
+}
+
+func (b DefaultBuildpackTOML) String() string {
+	return b.Buildpack.Name + " " + b.Buildpack.Version
+}
+
+func (b *DefaultBuildpackTOML) Build(bpPlan BuildpackPlan, config BuildConfig) (BuildResult, error) {
```

Comment:
This seemed to make things easier to test - since `config` is the same for every buildpack, it only needs to be created once. Then the tests can assert on the specific value of `bpPlan` without having to make further modifications. I can change it though, if you think it's better to have the plan nested.